VideoLocker/src/main/java/org/edx/mobile/module/db/impl/IDatabaseImpl.java

```diff
@@ -110,132 +110,176 @@ public Integer updateVideoAsOnlineByVideoId(String videoId, final DataCallback<I
-            String chapter, String section, final DataCallback<Boolean> callback) {
-        DbOperationExists op = new DbOperationExists(false,DbStructure.Table.DOWNLOADS, 
-                new String[] {DbStructure.Column.VIDEO_ID}, 
-                DbStructure.Column.SECTION + "=? AND " + DbStructure.Column.CHAPTER + "=? AND " 
-                        + DbStructure.Column.EID + "=? AND "+ DbStructure.Column.DOWNLOADED + "=? AND "
+                                               String chapter, String section, final DataCallback<Boolean> callback) {
+        DbOperationExists op = new DbOperationExists(false, DbStructure.Table.DOWNLOADS,
+                new String[]{DbStructure.Column.VIDEO_ID},
+                DbStructure.Column.SECTION + "=? AND " + DbStructure.Column.CHAPTER + "=? AND "
+                        + DbStructure.Column.EID + "=? AND " + DbStructure.Column.DOWNLOADED + "=? AND "
                         + DbStructure.Column.USERNAME + "=?",
-                        new String[] {section, chapter, enrollmentId, 
-                String.valueOf(DownloadedState.DOWNLOADING.ordinal()),username()}, null);
+                new String[]{section, chapter, enrollmentId,
+                        String.valueOf(DownloadedState.DOWNLOADING.ordinal()), username()}, null);
         op.setCallback(callback);
         return enqueue(op);
 
     }
 
     @Override
-    public List<Long> getDownloadingVideoDmIdsForSection(String enrollmentId, String chapter, 
-            String section , final DataCallback<List<Long>> callback ) {
-        DbOperationGetColumn<Long> op = new DbOperationGetColumn<Long>(true,DbStructure.Table.DOWNLOADS, 
+    public List<Long> getDownloadingVideoDmIdsForSection(String enrollmentId, String chapter,
+                                                         String section, final DataCallback<List<Long>> callback) {
+        DbOperationGetColumn<Long> op = new DbOperationGetColumn<Long>(true, DbStructure.Table.DOWNLOADS,
                 new String[]{DbStructure.Column.DM_ID},
-                DbStructure.Column.DOWNLOADED + "=? AND "+ DbStructure.Column.EID + "=? AND " 
+                DbStructure.Column.DOWNLOADED + "=? AND " + DbStructure.Column.EID + "=? AND "
                         + DbStructure.Column.CHAPTER + "=? AND "
                         + DbStructure.Column.SECTION + "=? AND "
                         + DbStructure.Column.USERNAME + "=?",
-                        new String[] {String.valueOf(DownloadedState.DOWNLOADING.ordinal()), 
-                enrollmentId, chapter, section, username()}, null, Long.class);
+                new String[]{String.valueOf(DownloadedState.DOWNLOADING.ordinal()),
+                        enrollmentId, chapter, section, username()}, null, Long.class);
         op.setCallback(callback);
         return enqueue(op);
     }
 
+    @Override
+    public Integer getDownloadingVideosCountForSection(String enrollmentId, String chapter,
```

Comment:
We should probably return primitives in these methods, as there is no point in returning objects. All the methods will need to be changed though, and if it's too much work then you can leave it for now I guess.